integration-cli/docker_cli_exec_test.go

```diff
@@ -125,10 +125,12 @@ func (s *DockerDaemonSuite) TestExecAfterDaemonRestart(c *check.C) {
 // Regression test for #9155, #9044
 func (s *DockerSuite) TestExecEnv(c *check.C) {
 	testRequires(c, DaemonIsLinux)
-	dockerCmd(c, "run", "-e", "LALA=value1", "-e", "LALA=value2",
+	out, _ := dockerCmd(c, "run", "-e", "LALA=value1", "-e", "LALA=value2",
 		"-d", "--name", "testing", "busybox", "top")
+	id := strings.TrimSpace(out)
+	c.Assert(waitRun(id), check.IsNil)
```

Comment:
if you use "testing" instead of an ID in the Assert then you don't need "out" or "id" and the dockerCmd above and below can remain unchanged.  IOW, just add `c.Assert(waitRun("testing"), check.IsNil)` to the original code, I think.